galpy/orbit/Orbits.py

```diff
@@ -5597,28 +5652,566 @@ def animate(self,*args,**kwargs): #pragma: no cover
+                    else 'false')
+        # Layout for multiple plots
+        if len(d1s) == 1:
+            xmin= [0,0,0]
+            xmax= [1,1,1]
+        elif len(d1s) == 2:
+            xmin= [0,0.55,0]
+            xmax= [0.45,1,1]
+        elif len(d1s) == 3:
+            xmin= [0,0.365,0.73]
+            xmax= [0.27,0.635,1]
+        # Colors
+        line_colors= ['#1f77b4', # muted blue
+                        '#ff7f0e', # safety orange
+                        '#2ca02c', # cooked asparagus green
+                        '#d62728', # brick red
+                        '#9467bd', # muted purple
+                        '#8c564b', # chestnut brown
+                        '#e377c2', # raspberry yogurt pink
+                        '#7f7f7f', # middle gray
+                        '#bcbd22', # curry yellow-green
+                        '#17becf'] # blue-teal
+        # When there are more than these # of colors needed, make up randoms
+        if self.size > len(line_colors):
+            line_colors.extend(["#%06x" % numpy.random.randint(0, 0xFFFFFF)
+                                for ii in range(self.size-len(line_colors))])
+        layout= """{{
+            scene:{{    xaxis: {{
+    title: '{xlabel}',
+    domain: [{xmin},{xmax}],
+    }},
+    yaxis: {{title: '{ylabel}'}},
+    zaxis: {{title: '{zlabel}'}},}},
+    margin: {{t: 20}},
+    hovermode: 'closest',
+    showlegend: false,
+    """.format(xlabel=xlabels[0],ylabel=ylabels[0],zlabel=zlabels[0],xmin=xmin[0],xmax=xmax[0])
+
+        layout+="""}"""
+        # First plot
+        setup_trace1= """
+    let trace1= {{
+        x: data.x1_0.slice(0,numPerFrame),
+        y: data.y1_0.slice(0,numPerFrame),
+        z: data.z1_0.slice(0,numPerFrame),
+        customdata: data.time,
+        hovertemplate: '<b>{name}</b>' + '<br><b>{xlabel}</b>: %{{x}}' + '<br><b>{ylabel}</b>: %{{y}}' + '<br><b>{zlabel}</b>: %{{z}}' + '<br><b>t (Gyr)</b>: %{{customdata:.5f}}',
+        mode: 'lines',
+        name: '',
+        line: {{
+        shape: 'spline',
+        width: 3.,
+        color: '{line_color}',
+        }},
+        type: "scatter3d",
+    }};
+
+    let trace2= {{
+        x: data.x1_0.slice(0,numPerFrame),
+        y: data.y1_0.slice(0,numPerFrame),
+        z: data.z1_0.slice(0,numPerFrame),
+        customdata: data.time,
+        hovertemplate: '<b>{name} (Current location)</b>' + '<br><b>{xlabel}</b>: %{{x}}' + '<br><b>{ylabel}</b>: %{{y}}' + '<br><b>{zlabel}</b>: %{{z}}' + '<br><b>t (Gyr)</b>: %{{customdata:.5f}}',
+        mode: 'lines',
+        name: '',
+        line: {{
+        shape: 'spline',
+        width: 8.,
+        color: '{line_color}',
+        }},
+        type: "scatter3d",
+    }};
+    """.format(line_color=line_colors[0], name=names[0], xlabel=xlabels[0], ylabel=ylabels[0], zlabel=zlabels[0])
+        traces_cumul= """trace1,trace2"""
+        # milkyway plane surface
+        mw_bg_surface = f"""let mw_bg = {{
```

Comment:
Woah! I think there's an assumption here that the scale is in kpc? But when the x/y/z are in internal units, it should be x `1/_ro`. The most robust way to check might be to do
```
'kpc' in labeldict.get(d1,r'\mathrm{No\ xlabel\ specified}'))
```
because you can't directly use `xlabel` in case it's been overwritten. Also see my comment on perhaps making the black part of the image transparent.